pkg/scanner/scan.go

```diff
@@ -140,7 +141,7 @@ type Scanner struct {
 // Driver defines operations of scanner
 type Driver interface {
 	Scan(ctx context.Context, target, artifactKey string, blobKeys []string, options types.ScanOptions) (
-		results types.Results, osFound ftypes.OS, err error)
+		results types.Results, osFound ftypes.OS, layersMetadata ftypes.LayersMetadata, err error)
```

Comment:
What if returning a new struct like `ScanReponse` or `types.Report`? IIUC, it will not break compatibility unless we change the protobuf interfaces.

Reply:
added scanResponse - https://github.com/aquasecurity/trivy/pull/8394/commits/945ddc38ad1b5b7f497aa9d5a6f8891e000db593